src/highscore_gui.cpp

```diff
@@ -189,7 +189,7 @@ struct HighScoreWindow : EndGameHighScoreBaseWindow {
 		this->SetupHighScoreEndWindow();
 		Point pt = this->GetTopLeft(ScaleSpriteTrad(640), ScaleSpriteTrad(480));
 
-		SetDParam(0, _settings_game.game_creation.ending_year);
+		SetDParam(0, _game_mode == GM_NORMAL ? _settings_game.game_creation.ending_year : _settings_newgame.game_creation.ending_year);
```

Comment:
I don't think it's possible to get to the highscore table from Scenario Editor, but just in case...

```suggestion
		SetDParam(0, _game_mode == GM_MENU ? _settings_newgame.game_creation.ending_year : _settings_game.game_creation.ending_year);
```